lib/rules/arrow-parens.js

```diff
@@ -51,13 +51,32 @@ module.exports = {
         const sourceCode = context.getSourceCode();
 
 
+
         /**
          * Determines whether a arrow function argument end with `)`
          * @param {ASTNode} node The arrow function node.
          * @returns {void}
          */
         function parens(node) {
-            const token = sourceCode.getFirstToken(node, node.async ? 1 : 0);
+            const isAsync = node.async;
+            const token = sourceCode.getFirstToken(node, isAsync ? 1 : 0);
+
+            /**
+             * Remove the parenthesis around a parameter
+             * @param {Fixer} fixer Fixer
+             * @returns {string} fixed parameter
+             */
+            function fixParamsWithParenthesis(fixer) {
+                const paramToken = context.getTokenAfter(token);
+                const closingParenToken = context.getTokenAfter(paramToken);
+                const asyncToken = isAsync ? context.getTokenBefore(token) : null;
```

Comment:
Can you use `sourceCode.getTokenBefore` and `sourceCode.getTokenAfter` instead of `context.getTokenBefore` and `context.getTokenAfter`? The latter two are deprecated.